### PR TITLE
[control-plane-manager][docs] Update documentation about managing K8s versions

### PR DESCRIPTION
**Pull request by z9r5**
## Description
Update documentation about managing versions.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [X] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries
```changes
section: control-plane-manager
type: chore
summary: Update documentation about managing versions.
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

File: modules/040-control-plane-manager/docs/README.md [hunk 1]
```diff
@@ -54,20 +54,23 @@ The `control-plane` nodes are scaled automatically using the `node-role.kubernet
...
+
+If the version specified for upgrading (the [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter) does not match the current version of control plane in the cluster, then a smart strategy for changing component versions is launched:
+- General remarks
+  - Updating in different NodeGroups is performed in parallel. Within each NodeGroup, nodes are updated sequentially, one at a time.
 - When upgrading:
-  - The upgrades are performed **sequentially**, one minor version at a time: 1.22 -> 1.23, 1.23 -> 1.24, 1.24 -> 1.25.
-  - You cannot proceed to the next version until all the `control plane` components have been successfully upgraded to the current one.
-  - The version to upgrade to can only be one minor version ahead of the kubelet versions on the nodes.
+  - The upgrading is performed in sequential steps, one minor version at a time: 1.22 -> 1.23, 1.23 -> 1.24, 1.24 -> 1.25.
```
Comment (yalosev) [inline modules/040-control-plane-manager/docs/README.md:68]: `The upgrade` or `upgrading` ?

File: modules/040-control-plane-manager/docs/README.md [hunk 2]
```diff
@@ -54,20 +54,23 @@ The `control-plane` nodes are scaled automatically using the `node-role.kubernet
...
+The update of the **patch version** of the control plane components (i.e. within the minor version, for example, from 1.24.3 to 1.24.8) occurs automatically together with the 
+update of the Deckhouse version. You can't manage patch version updates.
 
-The upgrade of a **minor version** of any control plane component is performed in a safe way. You just need to specify the desired minor version in the `control plane` settings. Deckhouse implements a smart strategy for changing the versions of `control plane` components if the desired version does not match the current one:
+Updating the **minor version** of control plane components (e.g. from `1.23.*` to `1.25.*`) can be controlled using the [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter. In the `kubernetesVersion` parameter, you can specify the automatic update mode (value `Automatic`) or the desired minor version of the control plane. The version of control plane that is used by default (with `kubernetesVersion: Automatic`), as well as a list of supported versions of Kubernetes can be found in the [documentation](../../supported_versions.html#kubernetes).
```
Comment (z9r5) [inline modules/040-control-plane-manager/docs/README.md:60]:
```suggestion
Upgrading the **minor version** of control plane components (e.g. from `1.23.*` to `1.25.*`) can be controlled using the [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter. In the `kubernetesVersion` parameter, you can specify the automatic update mode (value `Automatic`) or the desired minor version of the control plane. The version of control plane that is used by default (with `kubernetesVersion: Automatic`), as well as a list of supported versions of Kubernetes can be found in the [documentation](../../supported_versions.html#kubernetes).
```

File: modules/040-control-plane-manager/docs/README.md (hunk 1 above)
Comment (z9r5) [reply modules/040-control-plane-manager/docs/README.md:68]: Replace 'updating' with 'upgrading'.

I will as @kirkonru to help with the other suggestion

File: modules/040-control-plane-manager/docs/README.md [hunk 3]
```diff
@@ -54,20 +54,23 @@ The `control-plane` nodes are scaled automatically using the `node-role.kubernet
...
+If the version specified for upgrading (the [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter) does not match the current version of control plane in the cluster, then a smart strategy for changing component versions is launched:
+- General remarks
+  - Updating in different NodeGroups is performed in parallel. Within each NodeGroup, nodes are updated sequentially, one at a time.
 - When upgrading:
-  - The upgrades are performed **sequentially**, one minor version at a time: 1.22 -> 1.23, 1.23 -> 1.24, 1.24 -> 1.25.
-  - You cannot proceed to the next version until all the `control plane` components have been successfully upgraded to the current one.
-  - The version to upgrade to can only be one minor version ahead of the kubelet versions on the nodes.
+  - The upgrading is performed in sequential steps, one minor version at a time: 1.22 -> 1.23, 1.23 -> 1.24, 1.24 -> 1.25.
+  - At each step, the control plane version is updated first, followed by the kubelet update on the cluster nodes. 
+  - The update at each step occurs a maximum of one version higher than the minimum version of kubelet on the nodes.
```
Comment (kirkonru) [inline modules/040-control-plane-manager/docs/README.md:70]:
```suggestion
  - At each step, the control plane version is upgraded first, followed by kubelet upgrades on the cluster nodes. 
  - On top of that, the maximum version increment is one at a time; that is, the new minor version can differ from the lowest kubelet version on the nodes by 1 at max.
```

File: modules/040-control-plane-manager/docs/README.md [hunk 4]
```diff
@@ -54,20 +54,23 @@ The `control-plane` nodes are scaled automatically using the `node-role.kubernet
...
 - When downgrading:
-  - The downgrade is performed **sequentially**, one minor version at a time: 1.25 -> 1.24, 1.24 -> 1.23, 1.23 -> 1.22.
-  - Master nodes cannot have a lower version than workers: the downgrade isn't possible if the kubelet versions on the nodes aren't downgraded yet.
-  - When downgrading, the component version can only be one version behind the highest ever used minor version of the control plane components:
-    - Suppose, `maxUsedControlPlaneVersion = 1.20`. In this case, the lowest possible version of the control plane components is `1.19`.
-
-[List of supported Kubernetes versions...](../../supported_versions.html#kubernetes)
+  - A successful downgrade is guaranteed only one version down from the maximum minor version of the control plane ever used in the cluster.
+  - The kubelet is downgraded on the cluster nodes first, followed by the control plane components downgrade.
```
Comment (kirkonru) [inline modules/040-control-plane-manager/docs/README.md:73]:
```suggestion
  - Successful downgrading is only guaranteed for a single version down from the maximum minor version of the control plane ever used in the cluster.
  - kubelets on the cluster nodes are downgraded first, followed by the control plane components.
```

File: modules/040-control-plane-manager/docs/README.md (hunk 3 above)
Comment (kirkonru) [reply modules/040-control-plane-manager/docs/README.md:70]: It seems to me that line 70 is not much different from line 68.

**Review by kirkonru — APPROVED** [review]

File: modules/040-control-plane-manager/docs/README.md [hunk 5]
```diff
@@ -54,20 +54,23 @@ The `control-plane` nodes are scaled automatically using the `node-role.kubernet
 
 ## Version control
 
-The **patch version** of any control plane component (i.e. within the minor version, for example, from 1.19.3 to 1.19.8) is upgraded automatically along with the Deckhouse version.
+The update of the **patch version** of the control plane components (i.e. within the minor version, for example, from 1.24.3 to 1.24.8) occurs automatically together with the 
+update of the Deckhouse version. You can't manage patch version updates.
```
Comment (z9r5) [inline modules/040-control-plane-manager/docs/README.md:58]:
```suggestion
**Patch versions** of control plane components (i.e. within the minor version, for example, from 1.19.3 to 1.19.8) are upgraded automatically together with the Deckhouse version updates. You can't manage patch version upgrades.
```

File: modules/040-control-plane-manager/openapi/doc-ru-config-values.yaml [hunk 6]
```diff
@@ -79,10 +79,10 @@ properties:
         properties:
           webhookURL:
             description: |
-              URL авторизациионного webhook'а.
+              URL
```
Comment (z9r5) [inline modules/040-control-plane-manager/openapi/doc-ru-config-values.yaml:82]:
```suggestion
              URL webhook'а авторизации.
```

File: modules/040-control-plane-manager/docs/README_RU.md [hunk 7]
```diff
@@ -54,20 +54,22 @@ title: Управление control plane
...
+Если указанная для обновления версия (параметр [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion)) не соответствует текущей версии control plane в кластере, то запускается умная стратегия изменения версий компонентов:
+- Общие замечания
+  - Обновление в разных NodeGroup выполняется параллельно. Внутри каждой NogeGroup узлы обновляются последовательно, по одному.
 - При upgrade:
-  - Обновление происходит **последовательно**, по одной минорной версии: 1.22 -> 1.23, 1.23 -> 1.24, 1.24 -> 1.25.
-  - Переход на следующий шаг невозможен до тех пор, пока все компоненты `control-plane` не обновились до текущего шага обновления.
-  - Обновление происходит максимум на одну версию новее, чем версии kubelet на узлах.
+  - Обновление происходит **последовательными этапами**, по одной минорной версии: 1.22 -> 1.23, 1.23 -> 1.24, 1.24 -> 1.25.
+  - На каждом этапе сначала обновляется версия control plane, затем — обновление kubelet на узлах кластера.  
+  - Обновление на каждом этапе происходит максимум на одну версию выше, чем минимальная версия kubelet на узлах.
```
Comment (z9r5) [inline modules/040-control-plane-manager/docs/README_RU.md:69]: @yalosev Мы можешь убрать эту строку? Вроде как она почти дублирует выражение "Обновление происходит **последовательными этапами**, по одной минорной версии". Есть нюанс, но вроде выглядит несущественным, и возможно лучше упростить?

File: modules/040-control-plane-manager/docs/README.md [hunk 8]
```diff
@@ -54,20 +54,22 @@ The `control-plane` nodes are scaled automatically using the `node-role.kubernet
...
+Upgrading **minor versions** of control plane components (e.g. from `1.23.*` to `1.25.*`) can be managed using the [`kubernetesVersion`](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter. It specifies the automatic update mode (if set to `Automatic`) or the desired minor version of the control plane. The default control plane version (to use with `kubernetesVersion: Automatic`) as well as a list of supported Kubernetes versions can be found in [the documentation](../../supported_versions.html#kubernetes).
+
+The control plane upgrade is performed in a safe way for both single-master and multi-master clusters. The API server may be temporarily unavailable during the upgrade. At the same time, it does not affect the operation of applications in the cluster and can be performed without scheduling a maintenance window.
+
+If the version specified for upgrading (the [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter) does not match the current version of control plane in the cluster, then a smart strategy for changing component versions is launched:
```
Comment (kirkonru) [inline modules/040-control-plane-manager/docs/README.md:63]:
```suggestion
If the target version (set in the [kubernetesVersion](../../installing/configuration.html#clusterconfiguration-kubernetesversion) parameter) does not match the current control plane version in the cluster, a smart strategy for changing component versions is applied:
```

File: modules/040-control-plane-manager/docs/README_RU.md (hunk 7 above)
Comment (z9r5) [inline modules/040-control-plane-manager/docs/README_RU.md:69]:
```suggestion
```

File: modules/040-control-plane-manager/docs/README.md [hunk 9]
```diff
@@ -54,20 +54,22 @@ The `control-plane` nodes are scaled automatically using the `node-role.kubernet
...
+- General remarks
+  - Updating in different NodeGroups is performed in parallel. Within each NodeGroup, nodes are updated sequentially, one at a time.
 - When upgrading:
-  - The upgrades are performed **sequentially**, one minor version at a time: 1.22 -> 1.23, 1.23 -> 1.24, 1.24 -> 1.25.
-  - You cannot proceed to the next version until all the `control plane` components have been successfully upgraded to the current one.
-  - The version to upgrade to can only be one minor version ahead of the kubelet versions on the nodes.
+  - Upgrades are carried out sequentially, one minor version at a time: 1.22 -> 1.23, 1.23 -> 1.24, 1.24 -> 1.25.
+  - At each step, the control plane version is upgraded first, followed by kubelet upgrades on the cluster nodes. 
+  - On top of that, the maximum version increment is one at a time; that is, the new minor version can differ from the lowest kubelet version on the nodes by 1 at max.
```
Comment (z9r5) [inline modules/040-control-plane-manager/docs/README.md:69]:
```suggestion
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

File: modules/040-control-plane-manager/docs/CONFIGURATION_RU.md [hunk 10]
```diff
@@ -2,6 +2,6 @@
 title: "Управление control plane: настройки"
 ---
 
-Управление компонентами control plane кластера осуществляется с помощью модуля `control-plane-manager`, а параметры кластера, влияющие на управление control plane, берутся из данных первичной конфигурации кластера (параметр `cluster-configuration.yaml` Secret'а `d8-cluster-configuration` в пространстве имен `kube-system`), которое создается при установке.
+Некоторые параметры кластера, влияющие на управление control plane, также берутся из ресурса [ClusterConfiguration](installing/configuration.html#clusterconfiguration), который создается при установке Deckhouse.
```
Comment (z9r5) [inline modules/040-control-plane-manager/docs/CONFIGURATION_RU.md:5]:
```suggestion
Некоторые параметры кластера, влияющие на управление control plane, также берутся из ресурса [ClusterConfiguration](installing/configuration.html#clusterconfiguration).
```

File: modules/040-control-plane-manager/docs/README.md [hunk 11]
```diff
@@ -54,20 +54,21 @@ The `control-plane` nodes are scaled automatically using the `node-role.kubernet
 
 ## Version control
 
-The **patch version** of any control plane component (i.e. within the minor version, for example, from 1.19.3 to 1.19.8) is upgraded automatically along with the Deckhouse version.
+**Patch versions** of control plane components (i.e. within the minor version, for example, from 1.19.3 to 1.19.8) are upgraded automatically together with the Deckhouse version updates. You can't manage patch version upgrades.
```
Comment (z9r5) [inline modules/040-control-plane-manager/docs/README.md:57]:
```suggestion
**Patch versions** of control plane components (i.e. within the minor version, for example, from `1.23.3` to `1.23.5`) are upgraded automatically together with the Deckhouse version updates. You can't manage patch version upgrades.
```

File: modules/040-control-plane-manager/docs/README_RU.md [hunk 12]
```diff
@@ -54,20 +54,21 @@ title: Управление control plane
 
 ## Управление версиями
 
-Обновление **patch-версии** какого-либо компонента control-plane (т.е. в рамках минорной версии, например с 1.24.3 на 1.24.8) происходит автоматически вместе с версией Deckhouse.
+Обновление **patch-версии** компонентов control plane (т.е. в рамках минорной версии, например с 1.24.3 на 1.24.8) происходит автоматически вместе с обновлением версии Deckhouse. Управлять обновлением patch-версий нельзя.
```
Comment (z9r5) [inline modules/040-control-plane-manager/docs/README_RU.md:57]:
```suggestion
Обновление **patch-версии** компонентов control plane (т.е. в рамках минорной версии, например с `1.23.3` на `1.23.5`) происходит автоматически вместе с обновлением версии Deckhouse. Управлять обновлением patch-версий нельзя.
```

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [3816](https://github.com/deckhouse/deckhouse/pull/3816) to the branch [release-1.43](https://github.com/deckhouse/deckhouse/tree/release-1.43) successful!

**Pull request merged**